### PR TITLE
add readthedocs

### PR DESCRIPTION
**Pull request by RunningLeon**
## Motivation

Add readthedocs

EN Link https://lmdeploy.readthedocs.io/en/latest/
CN Link https://lmdeploy-zh-cn.readthedocs.io/zh_CN/latest/index.html

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repositories?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit tests to ensure the correctness.
3. If the modification has a dependency on downstream projects of a newer version, this PR should be tested with all supported versions of downstream projects.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

**Comment by lvhan028** [comment]
May rebase the latest main

File: docs/zh_cn/make.bat [hunk 1]
```diff
@@ -0,0 +1,35 @@
+@ECHO OFF
```
Comment (lvhan028) [inline docs/zh_cn/make.bat:1]: What's `make.bat` used for?

Comment (RunningLeon) [reply docs/zh_cn/make.bat:1]: It's for generating html on windows by running `./make.bat html`.

File: README.md [hunk 2]
```diff
@@ -1,6 +1,13 @@
 <div align="center">
   <img src="resources/lmdeploy-logo.png" width="450"/>
 
+[![docs](https://img.shields.io/badge/docs-latest-blue)](https://lmdeploy.readthedocs.io/en/latest/)
+[![badge](https://github.com/InternLM/lmdeploy/workflows/lint/badge.svg)](https://github.com/InternLM/lmdeploy/actions)
+[![codecov](https://codecov.io/gh/InternLM/lmdeploy/branch/main/graph/badge.svg)](https://codecov.io/gh/InternLM/lmdeploy)
```
Comment (lvhan028) [inline README.md:6]: codecov shows unknown. May remove it. Can we add a version badge?

File: docs/zh_cn/make.bat (hunk 1 above)
Comment (lvhan028) [reply docs/zh_cn/make.bat:1]: Is it a must?

File: docs/zh_cn/api.rst [hunk 3]
```diff
@@ -0,0 +1,19 @@
+lmdeploy.lite
+-------------
+.. automodule:: lmdeploy.lite
```
Comment (lvhan028) [inline docs/zh_cn/api.rst:3]: The APIs in readthedocs aren't rendered successfully

File: docs/zh_cn/make.bat (hunk 1 above)
Comment (RunningLeon) [reply docs/zh_cn/make.bat:1]: not a must, but is for windows users to check readthedocs on local machine.

File: README.md (hunk 2 above)
Comment (RunningLeon) [reply README.md]: yes

File: requirements/readthedocs.txt [hunk 4]
```diff
@@ -0,0 +1,4 @@
+mmengine
```
Comment (AllentDan) [inline requirements/readthedocs.txt:1]: there is a requirements.txt outside the requirements directory. Shall we move it inside?

**Review by AllentDan — APPROVED** [review]

**Pull request merged**